commandapi-platforms/commandapi-bukkit/commandapi-bukkit-core/src/main/java/dev/jorel/commandapi/config/ConfigGenerator.java

```diff
@@ -0,0 +1,129 @@
+				wasConfigUpdated = true;
+				tryCreateSection(config, path, sections);
+				config.set(path, commentedConfigOption.option());
+			}
+
+			// Update config option comments
+			// Comments are kinda stupid, some elements are apparently null elements
+			// Also, both, YamlConfiguration#getComments(String) and CommentedConfigOption#comments() return unmodifiable list
+			// which by itself apparently aren't able to be checked for equality by the equals() method
+			// As a result, we wrap them in new ArrayLists first to be able to compare them
+			List<String> existingComment = new ArrayList<>(existingConfig.getComments(path));
+			existingComment.removeIf(Objects::isNull);
+			List<String> defaultComment = new ArrayList<>(commentedConfigOption.comment());
+
+			if (!existingComment.equals(defaultComment)) {
+				wasConfigUpdated = true;
+			}
+			config.setComments(path, commentedConfigOption.comment());
+		}
+		if (shouldRemoveValues) {
+			wasConfigUpdated = true;
+		}
+		return (wasConfigUpdated) ? process(config.saveToString()) : null;
+	}
+
+	private static YamlConfiguration process(String configAsString) throws InvalidConfigurationException {
+		String[] configStrings = configAsString.split("\n");
+		StringBuilder configBuilder = new StringBuilder();
+		for (String configString : configStrings) {
+			configBuilder.append(configString).append("\n");
+			if (!configString.contains("#")) {
+				configBuilder.append("\n");
+			}
+		}
+		YamlConfiguration config = new YamlConfiguration();
+		config.loadFromString(configBuilder.toString());
+		return config;
+	}
+
+	private static void tryCreateSection(YamlConfiguration config, String path, Set<String> existingSections) {
+		if (path.contains(".")) {
+			// We have to create a section, or multiple if applicable, first, if it doesn't exist already
+			String[] sectionNames = path.split("\\.");
+			// The last value is the config option
+			for (int i = 0; i < sectionNames.length - 1; i++) {
+				String sectionName = sectionNames[i];
+				if (!existingSections.contains(sectionName)) {
+					config.createSection(sectionName);
+					for (CommentedConfigOption<?> commentedSection : DefaultedBukkitConfig.ALL_SECTIONS) {
+						if (commentedSection.path().equals(sectionName)) {
+							config.setComments(sectionName, commentedSection.comment());
+						}
+					}
+				}
+				existingSections.add(sectionName);
+			}
+		}
+	}
+
+	private static boolean shouldRemoveOptions(YamlConfiguration config) {
+		Set<String> configOptions = config.getKeys(true);
+		List<String> sections = new ArrayList<>();
+		for (String configOption : configOptions) {
+			ConfigurationSection section = config.getConfigurationSection(configOption);
+			if (section != null) {
+				sections.add(configOption);
+			}
+		}
+		for (String sectionName : sections) {
+			configOptions.remove(sectionName);
+		}
+		List<String> defaultConfigOptions = DefaultedBukkitConfig.ALL_OPTIONS.stream().map(option -> option.path()).toList();
```

Comment:
- Don't use a stream here
- We're only ever using this to check if this collection contains a value, so we should be using a `Set` instead

Reply:
> - Don't use a stream here

Alright, rip the one-liner :smiling_face_with_tear: